config.yaml

```diff
@@ -25,7 +25,24 @@ teracy-dev-k8s:
   vm_gui: false
   vm_memory: 2048
   vm_cpus: 1
-  os: ubuntu # coreos-stable | coreos-alpha | coreos-beta | centos | opensuse | opensuse-tumbleweed
+
+  # SUPPORTED_OS:
```

Comment:
let's move this supported_os as dynamic supported config that users can configure.

the program will look up the specified "os" specified by users into the supported_os to configure the program properly.

For example:

```
supported_os:
  coreos-stable:
    box: coreos-stable
    bootstrap_os: coreos
    user: core
    box_url: "%{k8s_core_os_stable_box_url}"
```

`k8s_core_os_stable_box_url` should be defined as a variable setting